lib/network/http/utils.js

```diff
@@ -0,0 +1,109 @@
+'use strict'; // eslint-disable-line
+
+const errors = require('../../errors');
+
+/**
+ * Parse the Range header into an object
+ *
+ * @param {String} rangeHeader - The 'Range' header value
+
+ * @return {Object} object containing a range specification, with
+ * either of:
+ * - start and end attributes: a fully specified range request
+ * - a single start attribute: no end is specified in the range request
+ * - a suffix attribute: suffix range request
+ * - an error attribute of type errors.InvalidArgument if the range
```

Comment:
Any test you know of to validate this is good.

I believe AWS rules are those of HTTP RFC regarding range support. But those are sometimes not completely unambiguous when it comes to uncommon cases such as empty object support and suffix range.